zebra/zebra_srv6_vty.c

```diff
@@ -259,6 +260,346 @@ DEFUN (show_srv6_locator_detail,
+					"sid|behavior|context|daemons|locator|allocationType");
+			else
+				*json = ttable_json_with_json_text(
+					tt, "ssss",
+					"sid|behavior|context|daemons");
+		} else {
+			char *table;
+
+			table = ttable_dump(tt, "\n");
+			vty_out(vty, "%s\n", table);
+			XFREE(MTYPE_TMP, table);
+		}
+	}
+	ttable_del(tt);
+
+	if (!json && detail) {
+		vty_out(vty, "  Locator: %s\n",
+			sid_ctx->sid->locator ? sid_ctx->sid->locator->name
+					      : "-");
+		vty_out(vty, "  Allocation type: %s\n",
+			srv6_sid_alloc_mode2str(sid_ctx->sid->alloc_mode));
+	}
+}
+
+static void do_show_srv6_sid_all(struct vty *vty, json_object **json,
+				 struct srv6_locator *locator, bool detail)
+{
+	struct zebra_srv6 *srv6 = zebra_srv6_get_default();
+	struct zebra_srv6_sid_ctx *ctx;
+	struct listnode *node;
+	struct ttable *tt;
+
+	/* Prepare table. */
+	tt = ttable_new(&ttable_styles[TTSTYLE_BLANK]);
+	if (detail)
+		ttable_add_row(tt,
+			       "SID|Behavior|Context|Daemon/Instance|Locator|AllocationType");
+	else
+		ttable_add_row(tt, "SID|Behavior|Context|Daemon/Instance");
+	tt->style.cell.rpad = 2;
+	tt->style.corner = ' ';
+	ttable_restyle(tt);
+	ttable_rowseps(tt, 0, BOTTOM, true, '-');
+
+	for (ALL_LIST_ELEMENTS_RO(srv6->sids, node, ctx)) {
+		/* Skip contexts not associated with any SID */
+		if (!ctx->sid)
+			continue;
+
+		/* Skip SIDs from locators we are not interested in */
+		if (locator && ctx->sid->locator != locator)
+			continue;
+
+		do_show_srv6_sid_line(tt, ctx->sid, detail);
+	}
+
+	ttable_colseps(tt, 0, RIGHT, true, ' ');
+	ttable_colseps(tt, 1, LEFT, true, ' ');
+	ttable_colseps(tt, 1, RIGHT, true, ' ');
+	ttable_colseps(tt, 2, LEFT, true, ' ');
+	ttable_colseps(tt, 2, RIGHT, true, ' ');
+	ttable_colseps(tt, 3, LEFT, true, ' ');
+
+	/* Dump the generated table. */
+	if (tt->nrows > 1) {
+		if (json) {
+			if (detail)
+				*json = ttable_json_with_json_text(
+					tt, "ssssss",
+					"sid|behavior|context|daemons|locator|allocationType");
+			else
+				*json = ttable_json_with_json_text(
+					tt, "ssss",
+					"sid|behavior|context|daemons");
+		} else {
+			char *table;
+
+			table = ttable_dump(tt, "\n");
+			vty_out(vty, "%s\n", table);
+			XFREE(MTYPE_TMP, table);
+		}
+	}
+	ttable_del(tt);
+}
+
+DEFPY (show_srv6_sid,
+       show_srv6_sid_cmd,
+       "show segment-routing srv6 [locator NAME$locator_name] sid [X:X::X:X$sid_value] [detail$detail] [json]",
+       SHOW_STR
+       "Segment Routing\n"
+       "Segment Routing SRv6\n"
+       "Locator Information\n"
+       "Locator Name\n"
+       "SID\n"
+       "SID value\n"
+       "Detailed information\n"
+       JSON_STR)
+{
+	bool uj = use_json(argc, argv);
+	struct zebra_srv6 *srv6 = zebra_srv6_get_default();
+	struct srv6_locator *locator = NULL;
+	struct zebra_srv6_sid_ctx *sid_ctx = NULL, *c;
+	struct listnode *node;
+	json_object *json = NULL;
+
+	if (locator_name) {
+		locator = zebra_srv6_locator_lookup(locator_name);
+		if (!locator) {
+			if (uj == false)
```

Comment:
Usually, it's easier for an operator to get a valid JSON all the time when the operator expects it `{}` (empty) instead of just an empty string.